logstash-core/lib/logstash/pipeline.rb

```diff
@@ -64,24 +83,25 @@ def configure(setting, value)
     @settings[setting] = value
   end
 
-  def safe_filter_worker_count
-    default = @settings["default-filter-workers"]
-    thread_count = @settings["filter-workers"] #override from args "-w 8" or config
+  def safe_pipeline_worker_count
+    default = DEFAULT_SETTINGS[:default_pipeline_workers]
+    thread_count = @settings[:pipeline_workers] #override from args "-w 8" or config
     safe_filters, unsafe_filters = @filters.partition(&:threadsafe?)
+
     if unsafe_filters.any?
       plugins = unsafe_filters.collect { |f| f.class.config_name }
       case thread_count
       when nil
         # user did not specify a worker thread count
         # warn if the default is multiple
-        @logger.warn("Defaulting filter worker threads to 1 because there are some filters that might not work with multiple worker threads",
-          :count_was => default, :filters => plugins) if default > 1
+        @logger.warn("Defaulting pipeline worker threads to 1 because there are some filters that might not work with multiple worker threads",
+                     :count_was => default, :filters => plugins) if default > 1
```

Comment:
this is only a style comment but personally I think that if we are to break down parameters on multiple line, which I try to avoid, this multiline parameters construct is cleaner and flows better with the indentation.
```ruby
@logger.warn(
  "Defaulting pipeline worker threads to 1 because there are some filters that might not work with multiple worker threads", 
  :count_was => default, 
  :filters => plugins
) if default > 1
```
and if we consider that the `if` becomes non-obvious when placed last, we can always do
```ruby
default > 1 && @logger.warn(
  "Defaulting pipeline worker threads to 1 because there are some filters that might not work with multiple worker threads", 
  :count_was => default, 
  :filters => plugins
)
```